Add guide to 'plugin shading' in Minecraft

rainbowdashlabs opened the pull request:
A new file 'plugin-shading.md' has been introduced in the minecraft directory. It gives developers a solution on how to use the shadow plugin to include libraries which are not available on Maven Central when developing for Minecraft. The link for further guidance on shading dependency into a jar file is also attached. This document is added to fill the information gap on how to make a plugin independent from its dependencies when those are not listed on a standard repository.

rainbowdashlabs committed:
Add guide to 'plugin shading' in Minecraft

A new file 'plugin-shading.md' has been introduced in the minecraft directory.
It gives developers a solution on how to use the shadow plugin to include libraries which are not available on Maven Central when developing for Minecraft. The link for further guidance on shading dependency into a jar file is also attached. This document is added to fill the information gap on how to make a plugin independent from its dependencies when those are not listed on a standard repository.

Amejonah1200 reviewed (commented):
There is little to no information in this tag besides a redirect to your blog ._.

rainbowdashlabs commented:
> There is little to no information in this tag besides a redirect to your blog ._.

Yes because everything that is missing would be an example build file, which would need to be maintained again with the current shadow version. We would also end up with two sources of truth and we all know how well updating tags work c:

rainbowdashlabs committed:
Update Minecraft plugin-shading guide

Two informational sentences are added to 'plugin-shading.md' file in Minecraft. The updates include the necessity to relocate shaded dependencies to avoid conflicts with other plugins and a clarified direction to the link that explains how to shade dependencies into a jar file. This enhances the comprehensibility of the guide and emphasizes the importance of relocating dependencies.

Amejonah1200 commented on minecraft/plugin-shading.md:
```diff
@@ -0,0 +1,11 @@
+alias: ["gradle shading", "minecraft shading"]
+---
+
+Nehmen wir an unsere Library befindet sich nicht im Maven Central und wir können den Library Loader nicht verwenden.
```
```suggestion
Nehmen wir an unsere Library befindet sich nicht im Maven Central und wir können somit den Library Loader nicht verwenden.
```

rainbowdashlabs replied:
Switched to and/or to also take users into account on version which do not have the library loader

rainbowdashlabs committed:
Change wording
